### PR TITLE
Domains: Show auto renew button if purchased with credits

### PR DESCRIPTION
**Pull request by eltongo**
#### Changes proposed in this Pull Request

This PR enables the auto renew button even if there's no payment method (e.g. the purchase was done with credits). In that case, enabling the auto renew toggle will direct the user to update their payment method.

The auto-renew button should work with any purchase, as long as it's recurring.

#### Testing instructions

* For a domain/mapping purchased with credits, verify that the auto-renew toggle appears, and that it directs you to the payment method screen
* For domains or mappings purchased with a credit card, verify that the auto-renew toggle works as before, so that nothing is broken

### CODE REVIEW

**Comment by fditrapani** [comment]
> We might want to revisit this copy because I guess the circumstances can be different for not having a credit card on file.

Here's a suggestion for the copy:

> Title: Turn on auto-renew
> Body: We currently don't have any payment details on file for this domain to automatically renew it for you. Please add a payment method so we can continue your service without interruptions. 
> CTA: Add a payment method
> Secondary button: Cancel

What do you think? 

Something we'll need to include in this PR is events to track button clicks. Let me know if you need any help coming up with names. We should track the toggle click as well as the button clicks in the modal.

**Comment by fditrapani** [comment]
Another quick note, I thought I should share. After removing the credit card, I confirmed that the auto-renew was disabled again. Thank you!

**Comment by daledupreez** [comment]
Also, it may be worth rebasing from master -- I had some trouble with customer home on my local environment, which may be what's affecting the tests.

File: client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-payment-method-dialog/index.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { Dialog } from '@automattic/components';
+import { withLocalizedMoment } from 'components/localized-moment';
```
[the remark this answers is not in the document]
Comment (eltongo) [reply client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-payment-method-dialog/index.jsx:12]: Just a leftover really. Thanks for noticing

File: client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-payment-method-dialog/index.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class AutoRenewPaymentMethodDialog extends Component {
+	static propTypes = {
+		isVisible: PropTypes.bool,
+		translate: PropTypes.func.isRequired,
+		purchase: PropTypes.object.isRequired,
```
[the remark this answers is not in the document]
Comment (eltongo) [reply client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-payment-method-dialog/index.jsx:23]: Nope. I'll add them

File: client/me/purchases/manage-purchase/auto-renew-toggle/index.jsx [hunk 3]
```diff
@@ -150,9 +199,17 @@ class AutoRenewToggle extends Component {
 		return isEnabled ? translate( 'Auto-renew (on)' ) : translate( 'Auto-renew (off)' );
 	}
 
+	shouldRender( purchase ) {
```
[the remark this answers is not in the document]
Comment (eltongo) [reply client/me/purchases/manage-purchase/auto-renew-toggle/index.jsx:202]: Good one. I removed the `isRechargeable` check, which isn't necessary now.

**Comment by eltongo** [comment]
> First on the future PR pile: when you're adding a credit card, clicking on the "<- Back" button in the "Add Credit Card" page goes to the "Manage Purchase" page instead of the domain settings page.

This and the redirect after the card is saved are a bit too complicated for this PR IMO, so I didn't bother with those, though we must address them.

**Pull request merged**